jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -75,9 +79,36 @@
       throws GeneralSecurityException {
     // Do not use {@link NetHttpTransport}. See {@link getConnectionFactory} for details.
     HttpTransport transport = new ApacheHttpTransport.Builder().doNotValidateCertificate().build();
+    addProxyCredentials(transport);
     return url -> new Connection(url, transport);
   }
 
+  private static void addProxyCredentials(HttpTransport transport) {
+    DefaultHttpClient httpClient =
+        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();
+
+    if (System.getProperty("http.proxyUser") != null) {
+      httpClient
+          .getCredentialsProvider()
+          .setCredentials(
```

Comment:
No, the http(s).proxyUser and http(s).proxyPassword are not picked up by default. You really need to use credentialsProvider or otherwise you will get 407.

Reply:
Cool. Thanks for figuring this out. We didn't know.

Reply:
@briandealwis looks like `proxyHost` and `proxyPort` are working, but `proxyUser` and `proxyPassword` have not been. I had tested that in https://github.com/GoogleContainerTools/jib/issues/1304#issuecomment-446783249. And after reading your comment, I also tested setting the user and the password on the command line, but it still doesn't work.

Reply:
Ah! My apologies — I missed that this was just user and password.